BraveShared/Analytics/UserReferralProgram.swift

```diff
@@ -180,6 +187,19 @@ public class UserReferralProgram {
         return nil
     }
     
+    /// Passing string, attempts to derive a ref code from it.
+    /// Uses very strict matching.
+    /// Returns the sanitized code, or nil if no code was found
+    public class func sanitize(input: String?) -> String? {
+        guard var input = input, input.hasPrefix(self.clipboardPrefix) else { return nil }
+        
+        input.removeFirst(self.clipboardPrefix.count + 1)
+        // Add any other potential validation here, e.g. validating the actual ref code string
+        
+        if input.isEmpty { return nil }
+        return input
```

Comment:
We can do `!input.isEmpty ? input : nil`

Reply:
Need to clear the clipboard when a match is found.

Reply:
We clear elsewhere. Will update @Brandon-T but not with the double negative 😄 
`input.isEmpty ? nil : input`